### PR TITLE
feature/1-initial-implementation

### LINKED ISSUE #1: Initial Implementation
Add an initial implementation

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #1

### CODE REVIEW

File: doc/developers_guide/developers_guide.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Developers Guide
+
+## Building the native image
+
+Current try to build a native image:
```
Comment (morazow) [inline doc/developers_guide/developers_guide.md:5]:
```suggestion
Currently we build a native image using the following command:
```

File: doc/developers_guide/developers_guide.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+```shell
+ ~/programme/graalvm-ce-java11-21.1.0/bin/native-image --static -cp target/parquet-edml-generator.jar com.exasol.edmlgenerator.parquet.Main --no-fallback
```
Comment (morazow) [inline doc/developers_guide/developers_guide.md:8]: You could also generalize the native-image path. 

Please also add if there are steps to install the native image.

File: src/main/java/com/exasol/edmlgenerator/parquet/Main.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import com.exasol.adapter.document.edml.EdmlDefinition;
+import com.exasol.adapter.document.edml.serializer.EdmlSerializer;
+import com.exasol.mavenprojectversiongetter.MavenProjectVersionGetter;
+
+import picocli.CommandLine;
+
+/**
+ * This class is the entrypoint for the CLI client.
+ */
+@CommandLine.Command(name = "parquet-edml-generator", versionProvider = Main.VersionProvider.class, mixinStandardHelpOptions = true, description = "This tool builds EDML mapping definitions for given parquet files.")
```
Comment (morazow) [inline src/main/java/com/exasol/edmlgenerator/parquet/Main.java:15]:
```suggestion
@CommandLine.Command(name = "parquet-edml-generator", versionProvider = Main.VersionProvider.class, mixinStandardHelpOptions = true, description = "This tool builds EDML mapping definitions for a given parquet file.")
```

File: src/main/java/com/exasol/edmlgenerator/parquet/ToUpperSnakeCaseConverter.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    /**
+     * Convert a string to UPPER_SNAKE_CASE.
+     * 
+     * @param inputString input string in camelCase, lower_snake_case or UPPER_SNAKE_CASE.
+     * @return string formatted in UPPER_SNAKE_CASE
+     */
+    static String toUpperSnakeCase(final String inputString) {
+        boolean wasPreviousUpperOrUnderscore = false;
```
Comment (morazow) [inline src/main/java/com/exasol/edmlgenerator/parquet/ToUpperSnakeCaseConverter.java:18]:
```suggestion
        boolean isPreviousUpperOrUnderscore = false;
```

**Review by morazow — CHANGES_REQUESTED** [review]

File: doc/developers_guide/developers_guide.md (hunk 1 above)
Comment (jakobbraun) [reply doc/developers_guide/developers_guide.md:5]: well, it does not work yet :-/ 
I just wanted to note down my last try somewhere

**Review by morazow — APPROVED** [review]

**Pull request merged**